### PR TITLE
clang: update to 4.0.1 und fix building

### PR DESCRIPTION
**Pull request by holgern**
fix compilation

### CODE REVIEW

File: mingw-w64-clang/PKGBUILD [hunk 1]
```diff
@@ -73,6 +73,7 @@ source=(https://llvm.org/releases/${pkgver}/llvm-${pkgver}.src.tar.xz{,.sig}
         "0401-mingw-w64-hack-and-slash-fixes-for-libc.patch"
         "0501-minor-fixes-to-get-lldb-building.patch"
         "0502-hack-to-use-64-bit-time-for-mingw.patch"
+		"0503-fix-compile-error-TaskPool.patch"
```
Comment (Alexpux) [inline mingw-w64-clang/PKGBUILD]: Use spaces instead tabs

File: mingw-w64-clang/PKGBUILD [hunk 2]
```diff
@@ -247,8 +250,9 @@ build() {
   # https://github.com/martine/ninja/issues/932
 
   if [ "${_compiler}" == "gcc" ]; then
-    export CC='gcc'
-    export CXX='g++'
+    msg "using gcc as compiler"
+    #export CC='gcc'
```
Comment (Alexpux) [inline mingw-w64-clang/PKGBUILD]: Why you disable this exports? I think without this by default will be clang used if it installed already

File: mingw-w64-clang/PKGBUILD [hunk 3]
```diff
@@ -521,4 +525,4 @@ package_mingw-w64-x86_64-llvm(){
 
 package_mingw-w64-x86_64-polly(){
   package_polly
-}
+}
```
Comment (Alexpux) [inline mingw-w64-clang/PKGBUILD]: Add new line at the end

File: mingw-w64-clang/PKGBUILD (hunk 2 above)
Comment (mati865) [reply mingw-w64-clang/PKGBUILD]: It was just a fail-safe.

**Comment by mati865** [comment]
Add this https://github.com/Alexpux/MINGW-packages/pull/2647/files#diff-f8431e29bf404d2b6e837a3e6a7e6b00R339

File: mingw-w64-clang/PKGBUILD (hunk 2 above)
Comment (holgern) [reply mingw-w64-clang/PKGBUILD]: The package does not build using makepkg-mingw in the msys2 terminal.
But it builds completely when doing make in the mingw32 terminal.
So i think, that the wrong gcc from msys2 is used.
removing both export lines fixes this.

Comment (mati865) [reply mingw-w64-clang/PKGBUILD]: @holgern this is wrong, it **has** to be build with `makepkg-mingw` in msys2 shell.

**Comment by holgern** [comment]
export CC='gcc'
    export CXX='g++'
assure that gcc is used for compiling.
Then unset both variables prevent that a wrong gcc is used in the make process.

**Pull request closed without merging**